docs/_providers/hetzner.md

```diff
@@ -54,19 +54,60 @@ At this time you cannot update SOA records via DNSControl.
 
 ### Rate Limiting
 
-In case you are frequently seeing messages about being rate-limited:
+Hetzner is rate limiting requests in multiple tiers: per Hour, per Minute and
+ per Second.
+
+Depending on how many requests you are planning to perform, you can adjust the
+ delay between requests in order to stay within your quota.
+
+The setting `optimize_for_rate_limit_quota` controls this behavior and accepts
+ a case-insensitive value of
+- `Hour`
+- `Minute`
+- `Second`
+
+The default for `optimize_for_rate_limit_quota` is `Second`.
```

Comment:
I'm not sure I understand how the ratelimiting works but my concern is that the default should be something reasonable for the typical user.  It might be, I'm just unsure because I don't fully understand how it works.

Reply:
I think the lowest rate-limit is a reasonable one to start with and when the user hits the next level we will print a warning for them to consider upping it. Burning though 1000 requests [1] in less than an hour which regular use seems quite unlikely given that we now creating and updating records in batches.

Starting with the highest rate-limit `hour` (1 request every 3.6 seconds) may annoy users who never approach the limit due to small/infrequent changes to their zones.

[1] As I understand it, Hetzner is rate-limiting with multiple sliding windows (second, minute, hour). In my case they upped the limits to 2 requests per sec, 120 per min, 2000 per hour. From my testing before they upped the limits, I think the defaults were 1 per sec, 60 per minute (a guess, based on the fact that I could sustain 1 req/s though out the entire test suite) and likely 1000 per hour (another guess given that the other rates were doubled for me). This also suggests that the second and minute rates are identical for all users.